starport/services/scaffolder/module.go

```diff
@@ -201,48 +245,36 @@ func (s Scaffolder) ImportModule(tracer *placeholder.Tracer, name string) (sm xg
 	return sm, finish(s.path, s.modpath.RawPath)
 }
 
-func moduleExists(appPath string, moduleName string) (bool, error) {
-	absPath, err := filepath.Abs(filepath.Join(appPath, moduleDir, moduleName))
-	if err != nil {
-		return false, err
+func checkModuleName(appPath, moduleName string) error {
+	// go keyword
+	if token.Lookup(moduleName).IsKeyword() {
+		return fmt.Errorf("%s is a Go keyword", moduleName)
 	}
 
-	_, err = os.Stat(absPath)
-	if os.IsNotExist(err) {
-		// The module doesn't exist
-		return false, nil
+	// check if the name is a reserved name
+	if _, ok := reservedNames[moduleName]; ok {
+		return fmt.Errorf("%s is a reserved name and can't be used as a module name", moduleName)
 	}
 
-	return true, err
-}
-
-func checkModuleName(moduleName string) error {
-	// go keyword
-	if token.Lookup(moduleName).IsKeyword() {
-		return fmt.Errorf("%s is a Go keyword", moduleName)
+	// check if the name can imply potential store key collision
+	for _, defaultStoreKey := range defaultStoreKeys {
+		if strings.HasPrefix(moduleName, defaultStoreKey) {
+			return fmt.Errorf("the module name can't be prefixed with %s because of potential store key collision", defaultStoreKey)
+		}
 	}
 
-	// name of default registered module
-	switch moduleName {
-	case
-		"auth",
-		"genutil",
-		"bank",
-		"capability",
-		"staking",
-		"mint",
-		"distr",
-		"gov",
-		"params",
-		"crisis",
-		"slashing",
-		"ibc",
-		"upgrade",
-		"evidence",
-		"transfer",
-		"vesting":
-		return fmt.Errorf("%s is a default module", moduleName)
+	// check store key with user's defined modules
+	// we consider all user's defined modules use the module name as the store key
+	entries, err := os.ReadDir(filepath.Join(appPath, moduleDir))
+	if err != nil {
+		return err
 	}
+	for _, entry := range entries {
+		if strings.HasPrefix(moduleName, entry.Name()) {
```

Comment:
I now that this already silently checks it but to avoid any possible bugs in the feature (in case we refactor this func for ex.) can we run moduleExists func anyway?

Reply:
yes actually